From 98af401441b1977dd9a18f2fe4bfa18805599539 Mon Sep 17 00:00:00 2001
From: Jay Cornwall <Jay.Cornwall@amd.com>
Date: Wed, 27 Jul 2016 17:51:09 -0500
Subject: [PATCH 5256/5855] drm/amdkfd: Report legacy doorbell type for Tonga

HSA_CAP_DOORBELL_TYPE_1_0 is dependent on the ASIC feature
CP_HQD_PQ_CONTROL.SLOT_BASED_WPTR, which was not introduced in VI
until Carrizo.

Report HSA_CAP_DOORBELL_TYPE_PRE_1_0 on Tonga ASICs.

Change-Id: I20549b86d6d2eb4fd62e5ce4282c67f043e60a19
Signed-off-by: Jay Cornwall <Jay.Cornwall@amd.com>
---
 drivers/gpu/drm/amd/amdkfd/kfd_topology.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/amd/amdkfd/kfd_topology.c b/drivers/gpu/drm/amd/amdkfd/kfd_topology.c
index 67b9e16..24b1c2f 100644
--- a/drivers/gpu/drm/amd/amdkfd/kfd_topology.c
+++ b/drivers/gpu/drm/amd/amdkfd/kfd_topology.c
@@ -1107,12 +1107,12 @@ int kfd_topology_add_device(struct kfd_dev *gpu)
 	switch (dev->gpu->device_info->asic_family) {
 	case CHIP_KAVERI:
 	case CHIP_HAWAII:
+	case CHIP_TONGA:
 		dev->node_props.capability |= ((HSA_CAP_DOORBELL_TYPE_PRE_1_0 <<
 			HSA_CAP_DOORBELL_TYPE_TOTALBITS_SHIFT) &
 			HSA_CAP_DOORBELL_TYPE_TOTALBITS_MASK);
 		break;
 	case CHIP_CARRIZO:
-	case CHIP_TONGA:
 	case CHIP_FIJI:
 		pr_debug("amdkfd: adding doorbell packet type capability\n");
 		dev->node_props.capability |= ((HSA_CAP_DOORBELL_TYPE_1_0 <<
-- 
2.7.4

